src/atomic/atomic_intrinsics_msvc.hpp

```diff
@@ -52,6 +52,37 @@
 
 #include <assert.h>
 
+// 32-bit Windows compilation
+#ifndef _M_X64
+# pragma intrinsic(_InterlockedCompareExchange64)
+
+__inline __int64 _InterlockedExchangeAdd64(__int64 volatile *storage,
+                                           __int64 value) {
+  // Short circuit
+  if (value == 0) {
+    return *storage;
+  }
+
+  // Perform operation
+  __int64 old_value;
+  do {
+    old_value = *storage;
+  } while(_InterlockedCompareExchange64(storage, (*storage + value), *storage) != old_value);
```

Comment:
This should be: `_InterlockedCompareExchange64(storage, (old_value + value), old_value) != old_value`, otherwise, you could get a different value for each `*storage` as it's changed by another thread.